### ISSUE TITLE
Error: HTTP error 404: Not Found

### ISSUE THREAD

**Issue by hi117** [z411/trackma#340]
When syncing, I am getting a the following error:
`Error: HTTP error 404: Not Found`
No other infornmation is provided, the actual sync does not appear to complete also.

**Comment by Birdulon** [comment]
MAL's API is currently down.

**Comment by z411** [comment]
What site are you syncing with?

**Comment by z411** [comment]
What site are you syncing with?

**Comment by z411** [comment]
It seems they shutted it down due to a vulnerability, passwords need to be reset as well. You'll get this message if you try to log-in through the website.
![chrome_2018-05-24_12-45-25](https://user-images.githubusercontent.com/1988512/40499593-68ceb59c-5f50-11e8-87b4-bbe81a9ade84.png)

**Comment by z411** [comment]
https://myanimelist.net/forum/?topicid=1731860

**Comment by z411** [comment]
Since it's been a while since MAL is down I made this script to export your Trackma list (anime+MAL only) to a MyAnimeList XML compatible list. This way you can import it in other site that supports this format.

https://gist.github.com/z411/04d528ec5e557b1acfe4f2225a781cec

I tested it in Anilist and it works, tell me if there's any issue.

**Comment by Birdulon** [comment]
Great stuff @z411, only problems encountered were on anilist's end with not having entries for some shows. Might be worth adding the functionality to the account selection dialog(s).

Made a quick edit to extract my friends' lists - only applies to people using that branch, which is probably just me, but on the offchance anyone else is and sees this, here you go https://gist.github.com/Birdulon/b03c2d0a258dc3fa62cdef9b9211feaf

**Comment by aa-zz** [comment]
I rigged together a [version](https://github.com/aa-zz/trackma) that just uses the API calls that the website makes. Works well enough I guess...

**Comment by ahmubashshir** [comment]
Is it possible to use jikan.moe api for mal?

**Comment by z411** [comment]
@ahmubashshir Apparently jikan.moe only works to fetch information, not to edit your list. The call to get the seasons is cool, though.

**Comment by TheVexedGerman** [comment]
Does the MAL API going into open beta mean that MAL support is going to return in the foreseeable future?
https://myanimelist.net/forum/?topicid=1804432

**Comment by z411** [comment]
@TheVexedGerman Yes. But as someone mentioned in the thread the "open" beta doesn't seem to be an open beta, it's just a closed beta where developers in it are allowed to release their working implementations. I'm not in the beta so I don't know what the API looks like at all. We'll have to wait for the public release.

**Comment by lastweakness** [comment]
Is it possible to use the Jikan API for now and switch to the official API later?

**Comment by lastweakness** [comment]
> I rigged together a [version](https://github.com/aa-zz/trackma) that just uses the API calls that the website makes. Works well enough I guess...

This seems amazing... but seems like u had to abandon it. @z411 Does this work reliably enough?

**Comment by abhinavk99** [comment]
> 
> 
> Is it possible to use the Jikan API for now and switch to the official API later? Also, how do clients like Anyme and many other Android clients work? Scraping?

The Jikan API can only be used to get read-only information about anime and user lists. There's no capability for updating user lists unfortunately.

**Comment by DimT** [comment]
Seems like MAL API releases July 6.
[https://myanimelist.net/forum/?topicid=1849732](https://myanimelist.net/forum/?topicid=1849732)

**Comment by DatGurney** [comment]
hope this gets fixed soon. seems like taiga has already re-implemented the API

**Comment by toprak** [comment]
we are waiting for new implementation of the mal api.

**Comment by z411** [comment]
Hi, sorry for the lateness. I have been away from all kind of programming due to personal issues this year, but I just want to clear up that support for the new MAL API is planned and I will get to it as soon as I can.

**Comment by toprak** [comment]
@z411 Hope you have fixed your issues. It is good to hear~

**Comment by z411** [comment]
Just to update, today I finally started work on the new MyAnimeList API on the `new-mal-api` branch. Support for authentication (which I wasn't really sure how to implement) is now complete. The rest should be quick.

**Comment by z411** [comment]
Most basic support should be done; I merged the changes to the master branch. I haven't tested it thoroughly yet so there are bugs and error handling isn't done yet so I'll keep working on it, but it should be in a working state if anyone wants to test.

**Comment by gabman15** [comment]
It seems to be working for me (partially)! Though I can't seem to update anything from the app. Whenever I run upload on the program I get a 
`Data warning: Can't process <anime_name>, will leave unsynced.`
error

**Comment by z411** [comment]
What output do you get if you run Trackma with the -d command line argument? e.g. `trackma-qt -d`

**Comment by gabman15** [comment]
Getting this:
`libmal: Updating item Shoujo Shuumatsu Ryokou...
[D] libmal: POST data: b'num_episodes_watched=5'
[D] libmal: URL: https://api.myanimelist.net/v3/anime/35838/my_list_status
Data: Can't process Shoujo Shuumatsu Ryokou, will leave unsynced.
[D] Data: Info: Connection error: HTTP Error 400: Bad Request`

**Comment by z411** [comment]
Can confirm. Thank you. Will look into it in a minute.

**Comment by z411** [comment]
It's a bug upstream. Implemented a workaround in `https://github.com/z411/trackma/commit/83d942e50d542dcb3d7e7f093d460f3c5ae63947` - it should work.
